Git header

Kimjinho-k opened the pull request.

Kimjinho-k committed:
feat:make github header UI

by using HTML, CSS

Kimjinho-k committed:
feat: github page header UI completed

HTML,CSS를 이용하여 github page 헤더 작성

Kimjinho-k committed:
feat: applied styles using CSS

HTML을 사용해 github header를 구조화 하고 CSS를 사용해 스타일 적용

Kimjinho-k committed:
feat: Add for style CSS

CSS를 사용하여 navbar 메뉴들에 대한 포지션  세부조정과 각 종 효과를 적용

holahoon commented on style.css:
```diff
@@ -0,0 +1,80 @@
+.header  {
+   background-color: darkgrey;
+}
+
+.navbar{
+   
+   width: 980px;
+   margin: auto;   
+  
+}
+
```
이런 불필요한 공백은 빼주세요

Kimjinho-k replied:
넵^^~

holahoon commented on style.css:
```diff
@@ -0,0 +1,80 @@
+   margin-right: 20px;
+}
+
+.navbar_menu{
+   
+   display: flex;
```
여기 공백도요

holahoon commented on header.html:
```diff
@@ -0,0 +1,47 @@
+              />
+              <button type="submit">Go</button>
+            </form>
+            <div class="login">
+              <li><a href="#">Sign in</a></li>
+              <li class="li2"><a href="#">Sign up</a></li>
+            </div>
```
`ul`을 사용 안하시고 `div`안에 `li`를 사용하신 이유가 있나요?

Kimjinho-k replied:
Sorry, just no reason

holahoon reviewed (commented):
코멘트 봐주세요~

Kimjinho-k committed:
fix: modified Trailing Space Delete

delete trailing space to CSS file

Kimjinho-k committed:
fix: change the tag

change the tag(line 38) - <div> >> <ul>

Kimjinho-k committed:
reafactor: Add div class

노트북 화면과 데스크탑 모니터 화면에서 해당 페이지 헤더 부분이 다르게
표시 되어 수정을 위해 form 태그와 button 태그를 div 태그로 wrapping
했습니다.

Kimjinho-k committed:
refactor: new tag styled by CSS

header.html의 새로 추가한 div 태그 .search의 자식요소 form 태그에
display flex를 적용헀고 .login 클래스에 width 값을 100%로 지정헀습니다.

heejinlee07 commented on header.html:
```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html lang="ko">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>GitHub_div.div.header</title>
+    <link rel="stylesheet" href="style.css" />
+  </head>
+
```
`<Head>`영역에 favicon을 추가하면 좋습니다. 
일반적으로 모든 사이트에서는 favicon이 사용되고 있습니다. 
자세한 사용방법은 아래 링크의 글을 읽어보시기 바랍니다. 
favicon외에도 `<Head>`영역에 필수로 사용되는 태그들에 대한 정보가 있습니다.
그리고 `<Head>` 아래 빈 공백 한 줄은 삭제해주세요!

[https://developer.mozilla.org/ko/docs/Learn/HTML/Introduction_to_HTML/The_head_metadata_in_HTML#맞춤_아이콘_추가하기](https://developer.mozilla.org/ko/docs/Learn/HTML/Introduction_to_HTML/The_head_metadata_in_HTML#%EB%A7%9E%EC%B6%A4_%EC%95%84%EC%9D%B4%EC%BD%98_%EC%B6%94%EA%B0%80%ED%95%98%EA%B8%B0)

heejinlee07 commented on header.html:
```diff
@@ -0,0 +1,49 @@
+      <nav class="navbar">
+        <div class="navbar_left">
+          <div class="logo">
+            <img
+              src="https://heropcode.github.io/GitHub-Responsive/img/logo.svg"
+              alt="GItHub Logo"
+            />
+          </div>
+          <ul class="navbar_menu">
+            <li><a href="#">Personal</a></li>
+            <li><a href="#">Open source</a></li>
+            <li><a href="#">Business</a></li>
+            <li><a href="#">Explore</a></li>
+          </ul>
```
`<img>`태그에 대체 텍스트를 넣으신 점이 좋아요. SEO 최적화를 위해 항상 이미지 태그에는 대체 텍스트를 지정하는 습관을 들이세요. 또 메뉴를 `<ul>`,`<li>`로 시멘틱하게 구성한 점이 좋아요:)

heejinlee07 commented on header.html:
```diff
@@ -0,0 +1,49 @@
+              <button type="submit">Go</button>
+            </form>
+	   </div> 
+            <ul class="login">
+              <li><a href="#">Sign in</a></li>
+              <li class="li2"><a href="#">Sign up</a></li>
+            </ul>
```
시멘틱을 고려하여 <a>태그보다는 <button>으로 작성해주시는 것이 좋습니다. <a>태그는 아래와 같은 상황에 쓰입니다.

```
HTML <a> 요소(앵커 요소)는 href 특성을 통해 다른 페이지나 같은 페이지의 
어느 위치, 파일, 이메일 주소와 그 외 다른 URL로 연결할 수 있는 하이퍼링크를 만듭니다. 
<a> 안의 콘텐츠는 링크 목적지의 설명을 나타내야 합니다.
```

Kimjinho-k replied:
혹시 삭제 된 단어가 있을까요? 코멘트를 제대로 이해하지 못했습니다!..
~~태그보다는 ~~으로 부분이 삭제 된 것 같습니다!

heejinlee07 commented on header.html:
```diff
@@ -0,0 +1,49 @@
+          </ul>
+          <div class="search_bar">
+           <div class="search">
+            <form onsubmit="event.preventDefault();" role="search">
+              <input
+                id="search"
+                type="search"
+                placeholder="Search..."
+                autofocus
+                required
+              />
+              <button type="submit">Go</button>
+            </form>
```
이 경우는 굳이 form을 사용하지 않고 `<input>` 태그만 사용하셔도 무방합니다. form은 보통 실무에서 로그인이나 회원가입 상황에서 많이 사용됩니다.

heejinlee07 commented on style.css:
```diff
@@ -0,0 +1,81 @@
+}
+
+.navbar_left{
+   display: flex;
+   align-items: center;
+   display: flex;
+   align-items: center;
```
해당 코드가 두 번 작성되었네요. 쓰지 않는 코드는 삭제해주세요.
```
display: flex;
align-items: center;
```

heejinlee07 reviewed (commented):
안녕하세요 반갑습니다 😃
첫 리뷰를 작성하시느라 고생많으셨습니다.

전반적으로 시멘틱하게 잘 작성해주셨네요. css도 flexbox를 이용하여 레이아웃 배치를 
하신 점이 좋습니다:)

그리고, 현업에서는 보통 Conversation 란에 개발한 내용에 대해 기록하여 pr을 올리기 때문에
앞으로 pr 리퀘스트 요청 시 Conversation 란에 자신이 작업한 내용을 상세히 작성해주시면 좋습니다:)

Kimjinho-k committed:
refactoring: Add link tag

header.html <head>태그 내에 favicon을 링크 해주는 <link> 태그를 추가
하였습니다.

Kimjinho-k committed:
fix: delete form tag and wrap with new class

search class 안의 form 태그를 삭제하고css flex box 적용을위해 search_box
class를 생성해 input 태그를 cover하였습니다.

Kimjinho-k committed:
fix: change class name

header.html에서 search_form class가 삭제 되고 search_box class가 새로
생성 됨에 따라 search_form 에서 search_box로 class 명을 변경 하였습니다.

Kimjinho-k committed:
fix: Deduplication css property

navbar_left에 display 속성과 align-items 속성이 중복 되어
삭제 하였습니다.
